### DISCUSSION TITLE
Is there an event that catches on Route change ?

### DISCUSSION THREAD

**Discussion by BatuhanCakir** [expo/router#657] [Q&A]
Hi,
I have a feature that should warn the user if he wants to save the input before switching the tab. Is there an event that provides this information?

Thank you in advance

**Comment by wcastand** [comment]
you can at your root/layout add an useEffect on segments or pathname

```typescript
import {useSegments} from 'expo-router'

[...]

const segments = useSegments()

useEffect(() => {
 // would trigger on each route changes
}, [segments])

**Reply by designly1** [reply]
could also use `usePathname`, especially if you wanted to see what route your on and where you're going

**Reply by johnsonfash** [reply]
does not trigger in all cases
